hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,287 @@
+                .split(Pattern.quote("-"))[0]
+                .split(Pattern.quote("."));
+
+        if (versions.length >= 3) {
+            try {
+                return new int[] {
+                    SUPPORTED_STREAM_FILE_VERSION,
+                    Integer.parseInt(versions[0]),
+                    Integer.parseInt(versions[1]),
+                    Integer.parseInt(versions[2]),
+                };
+            } catch (final NumberFormatException e) {
+                System.err.printf(
+                        "createFileHeader :: Error when parsing protobuf version string  [%s] with exception : [%s]%n",
+                        hapiVersion, e);
+                throw new InvalidProtobufVersionException("Invalid hapi version string: " + hapiVersion);
+            }
+        } else {
+            System.err.printf("createFileHeader :: Error when parsing protobuf version string [%s]%n", hapiVersion);
+            throw new InvalidProtobufVersionException("Invalid hapi version string: " + hapiVersion);
+        }
+    }
+
+    private static SignatureFile.Builder createSignatureFile(final KeyPair sigKeyPair)
+            throws SignatureException, InvalidKeyException, NoSuchAlgorithmException, NoSuchProviderException {
+        final SignatureObject metadataSignature = generateSignatureObject(metadataStreamDigest.digest(), sigKeyPair);
+        final SignatureObject fileSignature = generateSignatureObject(streamDigest.digest(), sigKeyPair);
+        return SignatureFile.newBuilder().setFileSignature(fileSignature).setMetadataSignature(metadataSignature);
+    }
+
+    private static void generateSigRecordStreamFile(final File filePath, SignatureFile.Builder signatureFile)
+            throws IOException {
+        try (BufferedOutputStream output = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            output.write(RecordStreamType.getInstance().getSigFileHeader()[0]);
+            signatureFile.build().writeTo(output);
+        } catch (final IOException e) {
+            System.err.printf(
+                    "generateSigRecordStreamFile :: Failed to generate signature file for [%s] with exception : [%s]%n",
+                    filePath.getName(), e);
+            throw e;
+        }
+    }
+
+    private static void outputStreamDigest(int[] fileHeader, String recordFile) throws IOException {
+        try (final SerializableDataOutputStream dosMeta =
+                        new SerializableDataOutputStream(new HashingOutputStream(metadataStreamDigest));
+                final SerializableDataOutputStream dos = new SerializableDataOutputStream(
+                        new BufferedOutputStream(new HashingOutputStream(streamDigest)))) {
+
+            // parse record file
+            final Pair<Integer, Optional<RecordStreamFile>> recordResult =
+                    readMaybeCompressedRecordStreamFile(recordFile);
+            final var f = recordResult.getValue();
+            if (f.isPresent()) {
```

Comment:
This would be a good place to use `Optional`'s `ifPresent(Consumer).orElseThrow(Throwable)` supported idiom.  Code would look a bit cleaner without all the `.get()`s.

Reply:
We have another hasStartObjectRunningHash() check next. Just a style preference.